src/dynamic.ts

```diff
@@ -0,0 +1,66 @@
+import * as Color from "./color";
+import { Command } from "./command";
+
+import { HashCommand } from "./commands/hash";
+import { ModuleCommand } from "./commands/module";
+import { ObjectCommand } from "./commands/object";
+import { SlotCommand } from "./commands/slot";
+import { VersionCommand } from "./commands/version";
+import { TestCommand } from "./commands/test";
+import { CloseCommand } from "./commands/close";
+
+import * as c from "./const";
+import {COMMAND_POINTER} from "./const";
+
+export class Dynamic extends Command {
+
+    public name = "graphene";
+    public description = "The graphene-cli is a cross-platform command line tool for working with PKCS#11 devices";
+
+    constructor() {
+        super();
+
+        this.commands.push(new TestCommand(this));
+        this.commands.push(new CloseCommand(this));
+        this.commands.push(new VersionCommand(this));
+        this.commands.push(new ModuleCommand(this));
+        this.commands.push(new SlotCommand(this));
+        this.commands.push(new ObjectCommand(this));
+        this.commands.push(new HashCommand(this));
+    }
+
+    public async run(args: string[]): Promise<Command> {
+            var args2 = args.slice(2);
+            var bigArgs = [];
+            var indexes = [];
+            try {
+                for(let i=0;i<args2.length;i++){
+                    for(var cmd in COMMAND_POINTER){
```

Comment:
Misnamed and not an ideal way to do this. it would have to be updated anytime a command is added.